### PR TITLE
Fix Airgap EksaVersion

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cluster/builder.go [hunk 1]
```diff
@@ -154,6 +154,28 @@ func (b FileSpecBuilder) getBundles(manifestReader *manifests.Reader) (*releasev
...
+		// this shouldn't return an error at this point due to getBundles performing similar operations prior to this call
+		release, err := mReader.ReadReleaseForVersion(b.cliVersion.GitVersion)
+		if err != nil {
+			return nil, err
+		}
+		return release, nil
+	}
+
+	return &releasev1.EksARelease{
+		Date:              "2023-07-26 20:02:17.931339039 +0000 UTC",
```
Comment (vivek-koppuru) [inline pkg/cluster/builder.go:168]: Why is this date hardcoded?

File: pkg/cluster/builder.go [hunk 2]
```diff
@@ -154,6 +154,28 @@ func (b FileSpecBuilder) getBundles(manifestReader *manifests.Reader) (*releasev
 	return bundles.Read(b.reader, bundlesURL)
 }
 
+func (b FileSpecBuilder) getEksaRelease(mReader *manifests.Reader) (*releasev1.EksARelease, error) {
+	if b.bundlesManifestURL == "" {
```
Comment (abhinavmpandey08) [inline pkg/cluster/builder.go:158]: Will this be `""` in airgap environments?

File: pkg/cluster/builder.go [hunk 3]
```diff
@@ -154,6 +154,28 @@ func (b FileSpecBuilder) getBundles(manifestReader *manifests.Reader) (*releasev
...
+
+	return &releasev1.EksARelease{
+		Date:              "2023-07-26 20:02:17.931339039 +0000 UTC",
+		Version:           b.cliVersion.GitVersion,
+		Number:            1,
+		GitCommit:         "",
+		GitTag:            "",
+		BundleManifestUrl: "",
+		EksABinary:        releasev1.BinaryBundle{},
+		EksACLI:           releasev1.PlatformBundle{},
```
Comment (abhinavmpandey08) [inline pkg/cluster/builder.go:175]: Also, why we these values just empty? Do they not matter? If so, shouldn't the function just return the version?

**Review by abhinavmpandey08 — COMMENTED** [review]
I might have

File: pkg/cluster/builder.go (hunk 2 above)
Comment (tatlat) [reply pkg/cluster/builder.go:158]: "" is there if bundles-override is NOT set. You can see a similar check happening [here](https://github.com/aws/eks-anywhere/pull/6299/files/7aa3d4094a048ce0eeaf5b7c03c0d9a5bc002f9e#diff-5a573820e4737e0df734c87f7ff19388a4717953c43f7406572b012a52451338R150)

**Review by vivek-koppuru — APPROVED** [review]

**Review by g-gaston — COMMENTED** [review]
/lgtm

**Review by g-gaston — APPROVED** [review]

**Comment by g-gaston** [comment]
Forcing the merge here bc it's quite difficult to test just one branch of the new code with the current setup. In practice, the program will mostly never walk that path.

It's not worth exporting private methods to testing or complicate the code to tinker with the failure paths.

**Pull request merged**